tools/query-udevadm.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+'''
+Print a json with information about usb/udev devices. The devices are selected using
+an input --pattern. This program will return a json describing the usb devices according 
+to the command `udevadm info <device>`. Information is collected for each udev device 
+matching the input pattern.
+
+ex. `python3 query-udevadm.py --pattern=/dev/video*`
+`python3 query-udevadm.py --pattern=/dev/disk/by-id/* --indent=2`
+
+The patterns are matched according to the python glob module. This only provides support 
+for simple unix style patterns.
+
+See also, for example, (or similar):
+`udevadm info /dev/sda | sed -n 's/E: //p'`
+
+The json format is:
+{
+    "devices":[
+            {
+                "path":<pattern match path>
+                "companionDevice": <known/expected devices in companion>
+                "udevInfo":{
+                    <udev-attr>:<udev-value>,
+                    ...
+                }
+            },
+            ... 
+        ]
+    }
+}
+'''
+
+# TODO should we generally move os to subprocess in companion?
+# TODO python2 support
+import sys
+import subprocess
+import glob
+import argparse
+import json
+
+PARSER = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+PARSER.add_argument('--pattern',
+                    action="store",
+                    type=str,
+                    default="/dev/serial/by-id/*",
+                    help="Path to search for usb devices."
+                    )
+PARSER.add_argument('--indent',
+                    action="store",
+                    type=int,
+                    default=None,
+                    help="Indent level for json output formatting."
+                    )
+ARGS = PARSER.parse_args()
+
+companionFamiliarDevices = {
+    "Pixhawk Autopilot":
+    {
+        "ID_MODEL" : "PX4_FMU_v2.x",
+        "ID_SERIAL" : "3D_Robotics_PX4_FMU_v2.x_0"
+    },
+    "Pixhawk Autopilot Bootloader":
+    {
+        "ID_MODEL" : 'PX4_BL_FMU_v2.x',
+        "ID_SERIAL" : "3D_Robotics_PX4_BL_FMU_v2.x_0"
+    },
+    "Blue Robotics HD Low Light USB Camera":
+    {
+        "ID_VENDOR_ID" : "05a3",
```

Comment:
maybe we should put the vid and pid in a usb field inside the device, making possible to identify the connection type and more future proof.

Reply:
These are not necessarily usb devices, there are disks, csi cameras like the raspberry pi, i2c devices, etc.